README.md

```diff
@@ -39,6 +39,31 @@
 | -             | -             | `3333...`                 | 500 (Internal Server Error) | 
 | -             | -             | anything else             | return found                |      
 
+## Email verification data
+
+### Request passcode
+
+| email address                      | request passcode response | response error code       | 
+| ---------------------------------- | ------------------------- | ------------------------- |
+| `no_session_id@email.com`          | 401 (unauthorised)        | `NO_SESSION_ID`           |
+| `email_verified_already@email.com` | 409 (conflict)            | `EMAIL_VERIFIED_ALREADY`  |
+| `max_emails_exceeded@email.com`    | 403 (forbidden)           | `MAX_EMAILS_EXCEEDED`     |
+| `bad_email_request@email.com`      | 400 (bad request)         | `BAD_EMAIL_REQUEST`       |
+| `upstream_error@email.com`         | 502 (bad gateway)         | `UPSTREAM_ERROR`          |
+| anything else                      | 201 (created)             | -                         |
+
+### Verify passcode
+
+| passcode      | verify passcode response | response error code              |
+| ------------- | ------------------------ | -------------------------------- |
+| `BBBBBB`      | 401 (unauthorised)       | `NO_SESSION_ID`                  | 
+| `CCCCCC`      | 403 (forbidden)          | `MAX_PASSCODE_ATTEMPTS_EXCEEDED` |
+| `DDDDDD`      | 404 (not found)          | `PASSCODE_NOT_FOUND`             |
+| `FFFFFF`      | 404 (not found)          | `PASSCODE_MISMATCH`              | 
```

Comment:
what would be the difference between passcode mismatch and not found, I was of the view  if it is a mismatch it means it's not found

Reply:
the email verification service does a get on the passcode mongo database based upon the session id. If there is no document for the session id then a `PASSCODE_NOT_FOUND` error is returned. If there is a document for the session id but the passcode in the verify request doesn't match what was stored in mongo then we get a `PASSCODE_MISMATCH`